Packs/EWS/Integrations/EWSO365/EWSO365.py

```diff
@@ -0,0 +1,2082 @@
+    }
+    return readable_output, output, items_as_incidents
+
+
+def get_folder(client: EWSClient, folder_path, target_mailbox=None, is_public=None):
+    """
+    Retrieve a folder from the target mailbox or client mailbox
+    :param client: EWS Client
+    :param folder_path: folder path to retrieve
+    :param (Optional) target_mailbox: target mailbox
+    :param (Optional) is_public: is the folder public
+    :return:
+    """
+    account = client.get_account(target_mailbox)
+    is_public = client.is_default_folder(folder_path, is_public)
+    folder = folder_to_context_entry(
+        client.get_folder_by_path(folder_path, account=account, is_public=is_public)
+    )
+    readable_output = tableToMarkdown(f"Folder {folder_path}", folder)
+    output = {CONTEXT_UPDATE_FOLDER: folder}
+    return readable_output, output, folder
+
+
+def folder_to_context_entry(f):
+    """
+    Create a context entry from a folder response
+    :param f: folder response
+    :return: dict context entry
+    """
+    try:
+        f_entry = {
+            "name": f.name,
+            "totalCount": f.total_count,
+            "id": f.id,
+            "childrenFolderCount": f.child_folder_count,
+            "changeKey": f.changekey,
+        }
+
+        if "unread_count" in [x.name for x in Folder.FIELDS]:
+            f_entry["unreadCount"] = f.unread_count
+        return f_entry
+    except AttributeError:
+        if isinstance(f, dict):
+            return {
+                "name": f.get("name"),
+                "totalCount": f.get("total_count"),
+                "id": f.get("id"),
+                "childrenFolderCount": f.get("child_folder_count"),
+                "changeKey": f.get("changekey"),
+                "unreadCount": f.get("unread_count"),
+            }
+
+
+def mark_item_as_read(
+    client: EWSClient, item_ids, operation="read", target_mailbox=None
+):
+    """
+    Marks item as read
+    :param client: EWS Client
+    :param item_ids: items ids to mark as read
+    :param (Optional) operation: operation to execute
+    :param (Optional) target_mailbox: target mailbox
+    :return: Output tuple
+    """
+    marked_items = []
+    item_ids = argToList(item_ids)
+    items = client.get_items_from_mailbox(target_mailbox, item_ids)
+    items = [x for x in items if isinstance(x, Message)]
+
+    for item in items:
+        item.is_read = operation == "read"
+        item.save()
+
+        marked_items.append(
+            {
+                ITEM_ID: item.id,
+                MESSAGE_ID: item.message_id,
+                ACTION: "marked-as-{}".format(operation),
+            }
+        )
+
+    readable_output = tableToMarkdown(
+        f"Marked items ({operation} marked operation)", marked_items
+    )
+    output = {CONTEXT_UPDATE_EWS_ITEM: marked_items}
+    return readable_output, output, marked_items
+
+
+def get_item_as_eml(client: EWSClient, item_id, target_mailbox=None):
+    """
```

Comment:
BTW did you have option to test it?

Reply:
option?

Reply:
`ews-get-items-as-eml` command is included in test playbook?

Reply:
it isn't, tested it manually